gamemodes/terrortown/gamemode/server/sv_corpse.lua

```diff
@@ -319,115 +204,34 @@ function CORPSE.ShowSearch(ply, rag, isCovert, isLongRange)
-	local stime = 0
-
-	if rag.killer_sample then
-		stime = math.max(0, rag.killer_sample.t - CurTime())
-	end
-
-	-- build list of people this player killed
-	local kill_entids = {}
-	local ragKills = rag.kills
+	local sData = bodysearch.AssimilateSceneData(ply, rag, isCovert, isLongRange)
 
-	for i = 1, #ragKills do
-		local vicsid = ragKills[i]
-
-		-- also send disconnected players as a marker
-		local vic = player.GetBySteamID64(vicsid)
-
-		kill_entids[#kill_entids + 1] = IsValid(vic) and vic:EntIndex() or -1
-	end
-
-	local lastid = -1
-
-	if rag.lastid and ply:IsActive() and roleData.isPolicingRole then
-		-- if the person this victim last id'd has since disconnected, send -1 to
-		-- indicate this
-		lastid = IsValid(rag.lastid.ent) and rag.lastid.ent:EntIndex() or - 1
-	end
-
-	-- Send a message with basic info
-	net.Start("TTT_RagdollSearch")
-	net.WriteUInt(rag:EntIndex(), 16) -- 16 bits
-	net.WriteUInt(owner, 8) -- 128 max players. (8 bits)
-	net.WriteString(nick)
-	net.WriteColor(role_color)
-
-	net.WriteUInt(#eq, 16) -- Equipment (16 = max.)
-
-	for i = 1, #eq do
-		net.WriteString(eq[i])
-	end
-
-	net.WriteUInt(subrole, ROLE_BITS) -- (... bits)
-	net.WriteString(team)
-	net.WriteInt(c4, bitsRequired(C4_WIRE_COUNT) + 1) -- -1 -> 2^bits (default c4: 4 bits)
-	net.WriteUInt(dmg, 30) -- DMG_BUCKSHOT is the highest. (30 bits)
-	net.WriteString(wep)
-	net.WriteBit(hshot) -- (1 bit)
-	net.WriteInt(dtime, 16)
-	net.WriteInt(stime, 16)
-
-	net.WriteUInt(#kill_entids, 8)
+	-- only in mode 0 everyone can confirm by pressing E
+	if bodysearch.GetInspectConfirmMode() == 0 or sData.base.isPublicPolicingSearch then
+		-- only give credits if body is also confirmed
+		if not isCovert then
+			bodysearch.GiveFoundCredits(ply, rag, isLongRange, sData.searchUID)
+		end
 
-	for i = 1, #kill_entids do
-		net.WriteUInt(kill_entids[i], 8) -- first game.MaxPlayers() of entities are for players.
+		if GetConVar("ttt_identify_body_woconfirm"):GetBool() and DetectiveMode() and not isCovert then
+			CORPSE.IdentifyBody(ply, rag, sData.searchUID)
+		end
+	elseif ply:IsActiveShopper() and not ply:GetSubRoleData().preventFindCredits and not isCovert then
+		-- in mode 1 and 2 every active shopping role can take credits
+		bodysearch.GiveFoundCredits(ply, rag, isLongRange, sData.searchUID)
 	end
 
-	net.WriteUInt(lastid, 8)
-
-	-- Who found this, so if we get this from a detective we can decide not to
-	-- show a window
-	net.WriteUInt(ply:EntIndex(), 8)
-	net.WriteString(words)
+	-- cache credits of corpse here, AFTER one might has taken them
+	sData.credits = CORPSE.GetCredits(rag, 0)
 
-	net.WriteBit(isLongRange)
-
-	-- 133 + string data + #kill_entids * 8 + team + 1
-	-- 200 + ?
-
-	-- workaround to make sure only detective searches are added to the scoreboard
-	net.WriteBool(ply:IsActive() and roleData.isPolicingRole and not isCovert)
+	-- identifier so we know whether a ttt_confirm_death was legit
+	ply.searchID = sData.searchUID
```

Comment:
this is straight from old TTT. It is done to verify that the attempted confirmation is valid. To confirm a player you have to know the UID, otherwise it is ignored by the server. I'm not sure how to explain it in a comment without explaining the whole system and writing 30 lines of comments?